src/custom/hooks/useLoadingWithTimeout.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useState } from 'react'
+
+export default function useLoadingWithTimeout(isLoading: boolean, time: number) {
+  const [delayedLoad, setDelayedLoad] = useState(isLoading)
+
+  useEffect(() => {
+    let timeout: NodeJS.Timeout | undefined
+    // price is being queried
+    if (isLoading) {
+      // isn't currently showing cow or hasnt yet been updated here
+      // so we clear any running timeouts ready to clear local loading state
+      // and essentially reset them
+      clearTimeout(timeout as NodeJS.Timeout)
```

Comment:
Shouldn't you clear only if it's set?
What happens with this cast when it's `undefined`?

```suggestion
      if (timeout) {
        clearTimeout(timeout)
      }
```

Reply:
Same for all timeouts here